### PR TITLE
Add --local flag in mixer bundle create command

### LINKED ISSUE #713: Provide an option to create a local bundle without copying from upstream
Currently, during bundle create, Mixer copies the bundle definition from upstream if it exists. This can remain as the default behavior. But users should be able to explicitly create a new local bundle even if the name exists upstream. This can be done by introducing a `--local `flag for `mixer bundle create`.
E.g. `mixer bundle create os-core --local`
This should create an empty `os-core` definition file in the \<local-bundles> directory with only the header information. If the file already exists in the directory, do not overwrite it. Inform the user with a message `Bundle "os-core" already exists in local-bundles; skipping`

### PR DESCRIPTION
**Pull request by ashleshaAtrey**
mixer bundle create command with local flag can be used
to create an empty bundle file with just the definition
header information.

fixes #713

Signed-off-by: Ashlesha Atrey <ashlesha.atrey@intel.com>

### CODE REVIEW

File: builder/bundle_control.go [hunk 1]
```diff
@@ -769,19 +769,22 @@ func createBundleFile(bundle string, path string) error {
...
 	// Fetch upstream bundle files if needed
 	if err := b.getUpstreamBundles(); err != nil {
 		return err
 	}
 	var err error
 	for _, bundle := range bundles {
 		path, _ := b.getBundlePath(bundle)
+		localPath := filepath.Join(b.Config.Mixer.LocalBundleDir, bundle)
+		if _, err = os.Stat(localPath); err == nil {
+			fmt.Printf("Bundle %q already exists in %q; skipping\n", bundle, localPath)
```
Comment (reaganlo) [inline builder/bundle_control.go]: I like that you used `localpath` instead of just `b.Config.Mixer.LocalBundleDir` so the user can directly copy and edit.
Maybe just reword it to `fmt.Printf("Bundle %q already exists at %q; skipping\n", bundle, localPath)`

I might have another PR to keep the bundle command messages consistent.

File: builder/bundle_control.go [hunk 2]
```diff
@@ -769,19 +769,22 @@ func createBundleFile(bundle string, path string) error {
...
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (ashleshaAtrey) [reply builder/bundle_control.go]: isn't it better to check if err is nil, If some error do occurs, we can still try creating bundle files from other conditions?

[the remark this answers is not in the document]
Comment (reaganlo) [reply builder/bundle_control.go]: I meant after the` if _, err = os.Stat(localPath); err == nil {}` block, what if there was some other error that is `!os.IsNotExist(err)`

[the remark this answers is not in the document]
Comment (ashleshaAtrey) [reply builder/bundle_control.go]: okay got it. Will change it

File: mixer/cmd/bundles.go [hunk 3]
```diff
@@ -273,6 +275,7 @@ func init() {
 
 	bundleCreateCmd.Flags().BoolVar(&bundleCreateFlags.add, "add", false, "Add the bundle(s) to your mix")
 	bundleCreateCmd.Flags().BoolVar(&bundleCreateFlags.git, "git", false, "Automatically apply new git commit")
+	bundleCreateCmd.Flags().BoolVar(&bundleCreateFlags.local, "local", false, "Create a local bundle(s) for your mix")
```
[the remark this answers is not in the document]
Comment (reaganlo) [reply mixer/cmd/bundles.go]: This one?
